File: e2e/playwright-typescript/providers/ValidationProvider.ts
import { Page } from '@playwright/test';
import { AccessibilityValidator } from '@/interactions/validators/AccessibilityValidator';
import { LanguageValidator } from '@/interactions/validators/LanguageValidator';
import { TaskValidator } from '@/interactions/validators/TaskValidator';
import { TasksValidator } from '@/interactions/validators/TasksValidator';

export class ValidationProvider {
  readonly tasks: TasksValidator;
  readonly task: TaskValidator;
  readonly language: LanguageValidator;
  readonly accessibility: AccessibilityValidator;

  constructor(page: Page, _isMobile: boolean) {
    this.tasks = new TasksValidator(page);
    this.task = new TaskValidator(page);
    this.language = new LanguageValidator(page);
    this.accessibility = new AccessibilityValidator();
  }
}

File: e2e/playwright-python/providers/validation_provider.py
from playwright.sync_api import Page
from interactions.validators.tasks_validator import TasksValidator
from interactions.validators.task_validator import TaskValidator
from interactions.validators.language_validator import LanguageValidator
from interactions.validators.accessibility_validator import AccessibilityValidator


class ValidationProvider:
    def __init__(self, page: Page, is_mobile: bool):
        self.tasks = TasksValidator(page)
        self.task = TaskValidator(page)
        self.language = LanguageValidator(page)
        self.accessibility = AccessibilityValidator()
